src/pages/vault/api-docs/[[...page]].tsx

```diff
@@ -0,0 +1,33 @@
+import { generateStaticPaths, generateStaticProps } from 'layouts/docs/server'
```

Comment:
These page files are not too big, but they are super repetitive - the only diff is the `basePath` and the `mdx-components` import. I think we can probably further abstract, but also think it makes sense to tackle the hairier problem of MDX custom components first.

(Hoping to get to both in this PR, but IMO may make sense to drop this further abstraction task if resolving the MDX custom component challenges drags out).

Reply:
I think the boilerplate gets reduced a bit further with this: https://github.com/hashicorp/dev-portal/pull/72

Reply:
Oh, nice! 💯 I'll rebase off that work 👍